### PR TITLE
Sdk enumerate

### LINKED ISSUE #475: sdk: All Enumerates should return the IDs only

### PR DESCRIPTION
**Pull request by lpabon**
**What this PR does / why we need it**:
Allow clients to have more flexibility by only returning ids in the Enumerates.

**Which issue(s) this PR fixes** (optional)
Closes #475 
Part of #407

**Special notes for your reviewer**:
Do not review the generated files commit.

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -698,9 +698,9 @@ func (c *Cluster) ToStorageCluster() *StorageCluster {
 		NodeId: c.NodeId,
 	}
 
-	cluster.Nodes = make([]*StorageNode, len(c.Nodes))
+	cluster.NodeIds = make([]string, len(c.Nodes))
```
Comment (pault84) [inline api/api.go]: This is used only by the sdk? if so i'm ok with this change if not we can't do this.

File: api/api.pb.go [hunk 2]
```diff
@@ -1062,53 +1062,53 @@ func (m *Group) GetId() string {
 // swagger:model
 type VolumeSpec struct {
 	// Ephemeral storage
-	Ephemeral bool `protobuf:"varint,1,opt,name=ephemeral,proto3" json:"ephemeral,omitempty"`
```
Comment (pault84) [inline api/api.pb.go]: Why are all the proto3 references deleted? did we change versions? does it not matter?

Comment (lpabon) [reply api/api.pb.go]: Great question. It all depends who generated them before and after. Right before we 'release', we should agree which protoc version to use and generate using that.

File: api/api.go (hunk 1 above)
Comment (lpabon) [reply api/api.go]: Yeah, the StorageCluster is an SDK-only structure.

**Pull request merged**